sw/banshee/src/engine.rs

```diff
@@ -696,8 +697,10 @@ impl<'a, 'b> Cpu<'a, 'b> {
                     }
                 } else if (value as usize) < self.wake_up.len() {
                     self.num_sleep.fetch_sub(1, Ordering::Relaxed);
-                    self.wake_up[value as usize - self.engine.base_hartid]
-                        .fetch_max(self.state.cycle + 1, Ordering::Release);
+                    self.wake_up[value as usize].fetch_max(self.state.cycle + 1, Ordering::Release);
```

Comment:
This changes how the `wake_up` vector is interpreted. Shouldn't this change also be reflected in the `binary_wfi` and in the clint functions?

Reply:
`binary_wfi` already interprets `wake_up` as "offset-free" (entry `0` being the first hart, regardless of `base_hartid`). This is a bug that was not caught by the banshee `wfi` test. Do you agree?

https://github.com/pulp-platform/snitch/blob/fcc67a79f2d4edb31c1ddd17167fb75251d04951/sw/banshee/src/engine.rs#L914-L915

Reply:
I think this depends on how you handle it in software. If your `base_hartid` is 10, and you want to wake up the core with `hartid` 10, do you write the value `10` or `0` to the wake_up register? My assumption was that you write the `hartid` (10), so you have to remove the offset in Banshee. In your updated version, you'd have to write a zero, right? Meaning the software already removes the offset.

Reply:
Correct, the offset is removed in software because that is also how it is in RTL (at least in Snitch, i don't know about mempool). RTL wires bit 0 of the wakeup register to the first hart in the cluster, regardless of the base hartid

Reply:
Perfect. In this case, everything is good to go from my side. In MemPool, we don‘t have an offset.